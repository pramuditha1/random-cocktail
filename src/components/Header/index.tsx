import React from "react";
import ButtonComp from "../Button";
import { localization } from "../../locale";
import FavouriteItems from "../../containers/FavouriteItems";
import ShoppingCartIcon from "@mui/icons-material/ShoppingCart";
export interface HeaderPropTypes {
  title: string;
}
const Header: React.FC<HeaderPropTypes> = ({ title }) => {
  const [open, setOpen] = React.useState(false);
  return (
    <header className="flex items-center justify-between p-4 bg-gray-800 text-white">
      <div className="text-lg font-bold">{title}</div>
      <ButtonComp
        variant="contained"
        startIcon={<ShoppingCartIcon />}
        text={localization.favouriteButton}
        onClickHandler={() => setOpen(true)}
      />
      {/* FavouriteItems modal will open upon button click */}
      <FavouriteItems open={open} setOpen={setOpen} />
    </header>
  );
};

export default Header;
