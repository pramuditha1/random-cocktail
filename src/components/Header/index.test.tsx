import React from "react";
import { render } from "@testing-library/react";
import Header, { HeaderPropTypes } from "./index";

test("Header matches snapshot", () => {
  const mockProps: HeaderPropTypes = {
    title: "cocktail app",
  };

  const { asFragment } = render(<Header {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
