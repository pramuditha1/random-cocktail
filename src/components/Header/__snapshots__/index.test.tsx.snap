// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex items-center justify-between p-4 bg-gray-800 text-white"
  >
    <div
      class="text-lg font-bold"
    >
      cocktail app
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ShoppingCartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
          />
        </svg>
      </span>
      Favourite Items
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </header>
</DocumentFragment>
`;
